packages/asset/contracts/Asset.sol

```diff
@@ -326,6 +359,38 @@ contract Asset is
         recyclingAmounts[catalystTokenId] = amount;
     }
 
+    /// @notice Set a new URI for specific tokenid
+    /// @param tokenId The token id to set URI for
+    /// @param metadata The new uri for asset's metadata
+    function setTokenMetadata(
+        uint256 tokenId,
+        string memory metadata
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        _setURI(tokenId, metadata);
+    }
+
+    /// @notice Set a new base URI
+    /// @param baseURI The new base URI
+    function setBaseURI(
+        string memory baseURI
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        _setBaseURI(baseURI);
+    }
+
+    /// @notice returns full token URI, including baseURI and token metadata URI
+    /// @param tokenId The token id to get URI for
+    /// @return tokenURI the URI of the token
+    function uri(
+        uint256 tokenId
+    )
+        public
+        view
+        override(ERC1155Upgradeable, ERC1155URIStorageUpgradeable)
```

Comment:
do we actually need to extend ERC1155Upgradeable ? Since ERC1155URIStorageUpgradeable is extending ERC1155Upgradeable, I think we can just extend ERC1155URIStorageUpgradeable and remove this
https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/token/ERC1155/extensions/ERC1155URIStorageUpgradeable.sol#L16

Reply:
Sure I would look into it